### PR TITLE
DOC: Add ``numpy.lib`` docs page

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @rgommers,

This PR updates docs to include new submodule structure, mainly the renewed `lib` submodule.

It also adds short instruction for running Ruff rule. Good news - it runs out that "preview" rules can be run from (pip) installable Ruff, it just requires an additional option.

### CODE REVIEW

File: doc/source/numpy_2_0_migration_guide.rst [hunk 1]
```diff
@@ -228,6 +228,22 @@ setitem                 Use ``arr[index] = value`` instead.
 ======================  ========================================================
 
 
+Ruff plugin
+-----------
+
+All the changes that we covered in the previous sections can be automatically applied
+to the codebase with a dedicated Ruff rule
```
Comment (ngoldbaum) [inline doc/source/numpy_2_0_migration_guide.rst:235]:
```suggestion
to the codebase with the dedicated Ruff rule,
```

File: doc/source/numpy_2_0_migration_guide.rst [hunk 2]
```diff
@@ -228,6 +228,22 @@ setitem                 Use ``arr[index] = value`` instead.
...
+`NPY201 <https://docs.astral.sh/ruff/rules/numpy2-deprecation/>`_.
+
+You should install Ruff, version ``0.1.8`` or above, and add to your ``pyproject.toml``::
+
+    [tool.ruff.lint]
+    extend-select = ["NPY201"]
+    preview = true
+
+To learn more about preview mode see `Ruff docs <https://docs.astral.sh/ruff/preview/>`_.
```
Comment (ngoldbaum) [inline doc/source/numpy_2_0_migration_guide.rst:244]: when this is closed perhaps add an issue to remove this once the feature leaves preview mode

**Review by ngoldbaum — COMMENTED** [review]
Good catch! And nice that we don't need to wait for ruff. Just a couple very minor comments. Please look over the docs build once it finishes to see if there are any visual issues that need to be fixed or new warnings in the build output.

**Review by rgommers — APPROVED** [review]
Thanks Mateusz. LGTM modulo's Nathan's minor comments.

File: doc/source/numpy_2_0_migration_guide.rst (hunk 1 above)
Comment (mtsokol) [reply doc/source/numpy_2_0_migration_guide.rst]: Done!

**Comment by mtsokol** [comment]
> Please look over the docs build once it finishes to see if there are any visual issues that need to be fixed or new warnings in the build output.

I inspected docs parts that were modified here and they look fine. I added a few small visual changes to existing release note files to get rid of some docs build warnings.

**Comment by ngoldbaum** [comment]
Thanks @mtsokol!

**Pull request merged**